src/maybe.js

```diff
@@ -0,0 +1,36 @@
+var _curry3 = require('./_curry3');
+
+
+module.exports = (function() {
+  function returnThis() { return this; }
+
+  function Just(x) { this.value = x; }
+  Just.prototype = {
+    of: function _justof(x) { return new Just(x); },
+    map: function _justmap(f) {
+      return new Just(f(this.value));
+    },
+    ap: function _justap(m) {
+      return m.map(this.value);
+    },
+    chain: function _justchain(f) {
+      return f(this.value);
+    },
+    isNothing: false,
+    isJust: true
+  };
+
+  function Nothing() {}
+  Nothing.prototype = {
+    of: Just.prototype.of,
+    map: returnThis,
+    ap: returnThis,
+    chain: returnThis,
+    isNothing: true,
+    isJust: false
+  };
```

Comment:
I also think there would need to be a `Maybe` type in here that the others extend.  I really, really would not want to promote signatures with types like `Just a | Nothing`.  :smile: